numpy/_core/tests/test_dtype.py

```diff
@@ -96,6 +96,9 @@ def test_invalid_types(self):
             assert_raises(TypeError, np.dtype, 'q8')
             assert_raises(TypeError, np.dtype, 'Q8')
 
+        # Make sure negative-sized dtype raises an error
+        assert_raises(TypeError, np.dtype, 'S-1')
```

Comment:
can you also add tests for `U-1` and `V-1`?

Reply:
Hello, thanks for the feedback. 
Just added a new commit with these new tests.